lib/view_model/contact_list/contact_list_view_model.dart

```diff
@@ -16,11 +16,23 @@ class ContactListViewModel = ContactListViewModelBase
 abstract class ContactListViewModelBase with Store {
   ContactListViewModelBase(this.contactSource, this.walletInfoSource)
       : contacts = ObservableList<ContactRecord>(),
-        walletContacts = walletInfoSource.values
-            .where((info) => info.address?.isNotEmpty ?? false)
-            .map((info) => WalletContact(
-                info.address, info.name, walletTypeToCryptoCurrency(info.type)))
-            .toList() {
+        walletContacts = [] {
+    walletInfoSource.values.forEach((info) {
+      if (info.address?.isNotEmpty ?? false) {
+        if (info.addresses != null) {
+          info.addresses.forEach((address, label) {
+            walletContacts.add(WalletContact(
+                address,
+                info.name + ' ($label)',
+                walletTypeToCryptoCurrency(info.type)));
+          });
+        } else {
+          walletContacts.add(WalletContact(info.address, info.name,
+              walletTypeToCryptoCurrency(info.type)));
+        }
+      }
```

Comment:
Avoid nested conditions. Use "optional" operator for call forEach method on info.addresses (`info.addresses?.forEach(...)`).

Reply:
Need to rework that conditional, because that condition does not catch case when `address` is empty or null  but `addresses` are not empty.

Reply:
Need to set `info.addresses` to empty map as default value, then check does it not empty. Maybe will be better to remove `info.address` property at all and use `info.addresses.keys.first` or `info.addresses.values.first` (or last element if you wanna to show "last used") as address by default.